### PR TITLE
add infinite retry to notify emily

### PR DESCRIPTION
**Pull request by hozzjss**

### CODE REVIEW

File: src/util/use-emily-deposit.ts [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+    }) => {
+      const res = await fetch("/api/emilyDeposit", {
+        method: "POST",
+        headers: {
+          "Content-Type": "application/json",
+        },
+        body: JSON.stringify(params),
+      });
+      if (!res.ok) {
+        if (failureCount > 2) {
```
Comment (matteojug) [inline src/util/use-emily-deposit.ts:24]: qq: the `>2` is just to avoid spamming users too soon?

File: src/util/use-emily-deposit.ts [hunk 2]
```diff
@@ -0,0 +1,38 @@
+import { NotificationStatusType } from "@/comps/Notifications";
+import { useNotifications } from "@/hooks/use-notifications";
+import { useMutation } from "@tanstack/react-query";
+
+const expBackoff = (attempt: number) =>
+  Math.min(attempt > 1 ? 2 ** attempt * 1000 : 1000, 30 * 1000);
```
Comment (matteojug) [inline src/util/use-emily-deposit.ts:6]: nit: I guess this comes from the doc, but why do we need the ternary? also, nit
```suggestion
  Math.min(2 ** attempt, 30) * 1000;
```

File: src/comps/Deposit.tsx [hunk 3]
```diff
@@ -412,7 +408,12 @@ const DepositFlowConfirm = ({
           <SecondaryButton onClick={() => handlePrevClick()}>
             PREV
           </SecondaryButton>
-          <PrimaryButton onClick={() => handleNextClick()}>NEXT</PrimaryButton>
+          <PrimaryButton
+            disabled={isPendingNotifyEmily}
```
Comment (matteojug) [inline src/comps/Deposit.tsx:412]: Should we also disable `PREV` if there's a deposit in flight we are tracking?

Comment (hozzjss) [reply src/comps/Deposit.tsx:412]: makes sense

File: src/util/use-emily-deposit.ts (hunk 1 above)
Comment (hozzjss) [reply src/util/use-emily-deposit.ts:24]: yes

File: src/comps/Deposit.tsx (hunk 3 above)
Comment (hozzjss) [reply src/comps/Deposit.tsx:412]: done

**Review by aldur — APPROVED** [review]

**Pull request merged**